module/platform/Webots/src/Webots.cpp

```diff
@@ -404,9 +403,10 @@ namespace module::platform {
                             // Decode the protocol buffer and emit it as a message
                             char* payload = reinterpret_cast<char*>(buffer.data()) + sizeof(length);
 
+                            // Service the watchdog
+                            emit<Scope::WATCHDOG>(ServiceWatchdog<Webots>());
```

Comment:
After you deserialise, so if it keeps failing it will reset the connection (one line lower)